### PR TITLE
Allow hiding of launcher on Run and Debug title bar when debugging

### PR DESCRIPTION
**Pull request by gjsjohnmurray**
I was motivated to create this PR by the new option to put the debug toolbar in the Command Center. Even before this I often found myself clicking the green 'play' button in the title bar of the Run and Debug view when I intended to click the same button on the floating debug toolbar. With the floating toolbar I solved the problem by positioning it over that title bar, which I obviously cannot do after moving the toolbar to Command Center.

The PR adds a setting `debug.hideLauncherWhileDebugging` to hide the view title bar control when debugging, moving its functionality to the '...' menu just as already happens if the debug toolbar location is set to `docked`.

junk
(recording shows original setting name before it was inverted)

Pinging @roblourens for debugging and @jrieken for the new Command Center location for the debug toolbar.

### CODE REVIEW

File: src/vs/workbench/contrib/debug/browser/debug.contribution.ts [hunk 1]
```diff
@@ -452,7 +452,7 @@ configurationRegistry.registerConfiguration({
 		},
 		'debug.toolBarLocation': {
 			enum: ['floating', 'docked', 'commandCenter', 'hidden'],
-			markdownDescription: nls.localize({ comment: ['This is the description for a setting'], key: 'toolBarLocation' }, "Controls the location of the debug toolbar. Either `floating` in all views, `docked` in the debug view, `commandCenter` (requires `{0}`), or `hidden`.", '#window.commandCenter#', '#window.titleBarStyle#'),
+			markdownDescription: nls.localize({ comment: ['This is the description for a setting'], key: 'toolBarLocation' }, "Controls the location of the debug toolbar. Either `floating` in all views, `docked` in the debug view, `commandCenter` (requires `{0}`), or `hidden`.", '#window.commandCenter#'),
```
Comment (gjsjohnmurray) [inline src/vs/workbench/contrib/debug/browser/debug.contribution.ts:455]: This is cleanup of a redundant argument I spotted when copying the line for reuse in the new setting.

File: src/vs/workbench/contrib/debug/browser/debug.contribution.ts [hunk 2]
```diff
@@ -588,8 +588,13 @@ configurationRegistry.registerConfiguration({
 		},
 		'debug.enableStatusBarColor': {
 			type: 'boolean',
-			description: nls.localize('debug.enableStatusBarColor', "Color status bar when debugger is active"),
+			description: nls.localize('debug.enableStatusBarColor', "Color status bar when debugger is active."),
```
Comment (gjsjohnmurray) [inline src/vs/workbench/contrib/debug/browser/debug.contribution.ts:591]: Fixing a nit.

**Comment by gjsjohnmurray** [comment]
Please consider adding this to the October 2023 milestone.

**Comment by roblourens** [comment]
Sorry for the slow response @gjsjohnmurray, I'm not up to date on notifications right now. Are you saying that the problem for you is going for that button when you meant to click the 'continue' button?

**Comment by gjsjohnmurray** [comment]
@roblourens yes, exactly that.

**Comment by gjsjohnmurray** [comment]
Now that the 1.83.1 recovery has shipped please consider taking this

**Comment by gjsjohnmurray** [comment]
@roblourens any hope of this making October before endgame starts next week?

**Comment by roblourens** [comment]
We can take this, but can you leave the default behavior the same?

**Comment by gjsjohnmurray** [comment]
I have changed the setting to an opt-in and renamed it `debug.hideLauncherWhileDebugging`

**Comment by roblourens** [comment]
I mean, I'm fine with settings that are false by default too 😁 but we have a mix of "enable" and "disable" settings in vscode so this is fine

**Review by roblourens — APPROVED** [review]

**Review by joyceerhl — APPROVED** [review]

**Pull request merged**